inorbit_republisher/inorbit_republisher/republisher.py

```diff
@@ -173,38 +173,44 @@ def callback(msg, repub=repub):
     # for pub in pubs.values():
     #     pub.unregister()
 
-# """
-# Extracts a value from the given message using the provided getter function
-# """
-# def extract_value(msg, getter_fn):
-#     # TODO(adamantivm) Graceful handling of missing values to extract
-#     # TODO(adamantivm) Allow serialization of complex values
-#     val = getter_fn(msg)
-#     return val
 
-# """
-# Extracts several values from a given nested msg field and returns a dictionary of
-# <field, value> elements
-# """
-# def extract_values_as_dict(msg, mapping):
-#     values = {}
-#     base_getter_fn = attrgetter(mapping['field'])
-#     base_value = base_getter_fn(msg)
-#     fields = mapping.get('mapping_options', {}).get('fields')
-#     for field in fields:
-#         getter_fn = attrgetter(field)
-#         try:
-#             val = getter_fn(base_value)
-#             # genpy.Time values can't be serialized into JSON. convert them to seconds
-#             # TODO(diegobatt): Catch other datatypes
-#             if isinstance(val, genpy.Time):
-#                 val = val.to_sec()
-#             # TODO(diegobatt): Make it possible to use a different key than the field
-#             values[field] = val
-#         except AttributeError as e:
-#             rospy.logwarn('Couldn\'t get attribute %s: %s', field, e)
-#     filter_fn = mapping.get('mapping_options', {}).get('filter')
-#     return values if not filter_fn or eval(filter_fn)(values) else None
+"""
+Extracts a value from the given message using the provided getter function
+"""
+
+
+def extract_value(msg, getter_fn):
+    # TODO(adamantivm) Graceful handling of missing values to extract
+    # TODO(adamantivm) Allow serialization of complex values
+    val = getter_fn(msg)
+    return val
+
+
+"""
+Extracts several values from a given nested msg field and returns a dictionary of
+<field, value> elements
+"""
+
+
+def extract_values_as_dict(msg, mapping, node):
```

Comment:
I don't like passing the node to this function just to have the ability to log inside, fyi I'm aware of this ugly code

Reply:
I agree. Can you add a quick TODO? This will be solved when we use the class format

Reply:
Done